server/lib/topic_cache.py

```diff
@@ -154,7 +191,32 @@ def load_files(fpath_list: List[str], name_overrides: Dict,
 
 
 def load(name_overrides: Dict) -> Dict[str, TopicCache]:
-  topic_cache_map = {}
+  topic_cache_map: dict[str, TopicCache] = {}
   for dc, fpath_list in TOPIC_CACHE_FILES.items():
     topic_cache_map[dc] = load_files(fpath_list, name_overrides, dc)
+
+  if is_custom_dc():
+    custom_dc_topic_cache = _load_custom_dc_topic_cache(name_overrides)
+    if custom_dc_topic_cache:
+      topic_cache_map[DCNames.CUSTOM_DC.value] = custom_dc_topic_cache
+      topic_cache_map[DCNames.MAIN_DC.value]._merge(custom_dc_topic_cache)
```

Comment:
This may not be the right thing to do, e.g. for undata.  Should we maybe get what DC the topic-cache is for from the JSON itself?

Reply:
Are you implying we should not merge custom with main or that we should support `dc=<any>` vs a constant `dc=custom`?

Reply:
For UN, we should merge the topic-cache into `undata` not `main`.  So let the topic-cache builder make that choice?